docs/source/getting-started/tabular-data/data-types.ipynb

```diff
@@ -0,0 +1,210 @@
+    "preprocessing by both `.fill_null` and `.fill_nan` to impute both nulls and NaNs.\n",
+    "\n",
+    "## String\n",
+    "Strings have no domain descriptors, \n",
+    "but also take up the most space in memory and are the slowest to work with.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Categorical\n",
+    "\n",
+    "Categorical data appears to be string data, \n",
+    "but its underlying data representation is `pl.UInt32` indices into a set of string labels.\n",
+    "This results in much lower memory usage and a faster runtime.\n",
+    "These integer indices can be retrieved via the `.to_physical` expression.\n",
+    "\n",
+    "Unfortunately, there are two limitations to keep in mind:\n",
+    "\n",
+    "* OpenDP forbids expressions that may add or remove categories,\n",
+    "because this triggers a data-dependent [categorical remapping warning](https://docs.pola.rs/api/python/stable/reference/api/polars.exceptions.CategoricalRemappingWarning.html).\n",
+    "Side-effects like this do not satisfy differential privacy. \n",
+    "This means OpenDP rejects the use of categorical data in, for example, `.fill_null` and binary expressions.\n",
+    "* The encoding of categories typically assigns indices according to the order of records in the data.\n",
+    "Since revealing row ordering does not satisfy differential privacy, \n",
+    "OpenDP only allows categorical group-by columns when the encoding is data-independent.\n",
+    "The `.cut` expression, for example, has a data-independent encoding.\n",
+    "\n",
+    "The following code shows the latter limitation in practice:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "context_categorical = dp.Context.compositor(\n",
+    "    data=pl.LazyFrame([pl.Series(\"categorical\", [\"A\", \"B\", \"C\"] * 1000, dtype=pl.Categorical)]),\n",
+    "    privacy_unit=dp.unit_of(contributions=1),\n",
+    "    privacy_loss=dp.loss_of(epsilon=1.0, delta=1e-7),\n",
+    "    split_evenly_over=1,\n",
+    ")\n",
+    "\n",
+    "query = (\n",
+    "    context_categorical.query()\n",
+    "    .group_by(\"categorical\")\n",
+    "    .agg(dp.len())\n",
+    ")\n",
+    "try:\n",
+    "    query.release()\n",
+    "    assert False, \"unreachable, should have raised\"\n",
+    "except dp.OpenDPException as err:\n",
+    "    # the error we would expect to get:\n",
+    "    assert \"Categories are data-dependent\" in str(err)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Temporal\n",
+    "\n",
+    "OpenDP supports three kinds of temporal data types: `pl.Date`, `pl.Datetime` and `pl.Time`.\n",
+    "Datetimes may also store time zone information, which are considered part of the data domain \n",
+    "(all datetimes in a column must share the same time zone), \n",
```

Comment:
Once the data is ingest, it isn't possible to have different time zones, because Polars stores the time zone and time unit in the schema, and the data itself is just a vector of 64-bit ints. I'd imagine Polars would throw an error if the time zones don't match, but that's Polars' concern, not ours.